transcom/mymove
# B 20441

## [B-20441](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A979321)

## [Int Review](https://github.com/transcom/mymove/pull/13546)
## [Int Review #2](https://github.com/transcom/mymove/pull/13643)
## [Int Review #3](https://github.com/transcom/mymove/pull/13661)

## Summary

When creating a payment request the 'LockedPriceCents' field is now used to determine the price of the MS and CS payment service items.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Office application
2. Login as a TOO
3. Find a move that doesn't have approved shipments yet (just to ensure good data)
4. Approve the shipments and add some combination of MS and CS service items

NOTE: At this point I would recommend editing the db directly and altering the lockedPriceCents column in mto_service_items to a different value. This way you can validate that the value from this column is the one being used, and not the one from re_task_order_fees (which was used previously and will likely be an identical value in your local to lockedPriceCents, but has the ability to change in prod. Hence this task to lock the pricing down)

Can use the following queries to easily find the service item for MS/CS fee to change the locked price.

select moves.id from moves where moves.locator ='INSERT_MOVE_CODE_HERE';

### For MS use msi.re_service_id ='1130e612-94eb-49a7-973d-72f33685e551'
### For CS use msi.re_service_id ='9dc919da-9b66-407b-9f17-05c0f03fcb50'
select * from mto_service_items msi where msi.move_id = 'YOUR_ID_FROM_PREV_QUERY' and msi.re_service_id ='1130e612-94eb-49a7-973d-72f33685e551';

6. Log in to the Prime simulator
7. Find the move you made available by approving the first shipment
8. Create a payment request that includes some combination of MS and CS service items (and more if you want)
9. Verify that the value in the 'LockedPriceCents' field from the MTOServiceItem (converted to lockedPriceCents paymentServiceItemParam) is being used to populate the payment service items for MS and CS. This can be done through breakpoints in the counseling_services_pricer or management_service_pricer, or by checking in the payment_service_items db table.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
